### ISSUE TITLE
[10.2.0-RC1] Http().singleRequest() is sending query params cached from the last request

### ISSUE THREAD

**Issue by mushtaq** [akka/akka-http#3342]
**Steps**

- clone repo: https://github.com/mushtaq/akka-http-client-bug
- run QuickstartApp
- run test UserRoutesSpec
- server log prints 
```txt
http://127.0.0.1:8080/users?id=100
http://127.0.0.1:8080/users?id=100
```

**Expected**

- server should print

```txt
http://127.0.0.1:8080/users?id=100
http://127.0.0.1:8080/users
```

**Comment by raboof** [comment]
Thanks for the report and reproducer! That indeed looks wrong.

Looking with wireshark, it seems actually the client is sending the correct request, it is the server that seems to behave incorrectly.

Indeed the problem appears to be introduced between M1 and RC1.

We will look into this!

**Comment by raboof** [comment]
The offending change is d3b1322d4526171370935ec7b4648bc45fabc672 . I have to go now, but this is definitely something we want to fix ;)

**Comment by jrudolph** [comment]
Ugh, indeed!

**Comment by jrudolph** [comment]
Fix in #3352, embarrassing that I didn't think about those other fields at all. Mutable state, what could go wrong...

**Comment by mushtaq** [comment]
> Fix in #3352, embarrassing that I didn't think about those other fields at all. Mutable state, what could go wrong...

Mutable state is dangerous.... especially for people who have become too comfortable using the immutable data structures 😄

**Comment by jrudolph** [comment]
Fixed in #3352.
